backend/webapp/src/main/java/org/bonitasoft/web/designer/controller/upload/AssetUploader.java

```diff
@@ -0,0 +1,102 @@
+/**
+ * Copyright (C) 2015 Bonitasoft S.A.
+ * Bonitasoft, 32 rue Gustave Eiffel - 38000 Grenoble
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2.0 of the License, or
+ * (at your option) any later version.
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.bonitasoft.web.designer.controller.upload;
+
+import java.io.IOException;
+import java.util.Iterator;
+
+import org.bonitasoft.web.designer.model.WebResourceable;
+import org.bonitasoft.web.designer.model.asset.Asset;
+import org.bonitasoft.web.designer.model.asset.AssetType;
+import org.bonitasoft.web.designer.repository.Repository;
+import org.bonitasoft.web.designer.repository.AssetRepository;
+import org.bonitasoft.web.designer.repository.exception.NotFoundException;
+import org.bonitasoft.web.designer.rest.ErrorMessage;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.multipart.MultipartFile;
+
+public class AssetUploader<T extends WebResourceable> {
+
+    protected static final Logger logger = LoggerFactory.getLogger(AssetUploader.class);
+
+    private Repository<T> repository;
+    private AssetRepository<T> assetRepository;
+
+    public AssetUploader(Repository<T> repository, AssetRepository<T> assetRepository) {
+        this.repository = repository;
+        this.assetRepository = assetRepository;
+    }
+
+    public ResponseEntity<ErrorMessage> upload(MultipartFile file, String id, String type) {
```

Comment:
Why are you using directly ResponseEntity here ?
Would it be better to throw exception on error and let ResourceControllerAdvice do his job ? IMHO it would remove a lot of code from this methods (ErrorMessage creation, logs)